tendermint/src/validator.rs

```diff
@@ -103,14 +101,22 @@ impl Info {
 
     /// Verify the given signature against the given sign_bytes using the validators
     /// public key.
-    pub fn verify_signature(&self, sign_bytes: &[u8], signature: &[u8]) -> bool {
-        if let Some(pk) = &self.pub_key.ed25519() {
-            let verifier = Ed25519Verifier::from(pk);
-            if let Ok(sig) = ed25519::Signature::from_bytes(signature) {
-                return verifier.verify(sign_bytes, &sig).is_ok();
-            }
-        }
-        false
+    pub fn verify_signature(&self, sign_bytes: &[u8], signature: &Signature) -> Result<(), Error> {
```

Comment:
I made `signature` typed here, and changed it from returning a `bool` to a `Result`. I very strongly suggest doing this for a few reasons:

- `Result` is `#[must_use]`, so it significantly lowers the risk of ignoring signature validation by making most cases of it a compile-time error
- `Result` provides better type safety. Here's an example of a real-world signature validation bug which resulted from the use of combinators + `bool` instead of `Result`: https://github.com/libp2p/rust-libp2p/pull/1127/files

Reply:
Thanks for this one!